tools/refguide_check.py

```diff
@@ -854,6 +858,43 @@ def init_matplotlib():
         HAVE_MATPLOTLIB = False
 
 
+def check_dist_keyword_names():
```

Comment:
What would be the reason for putting it here? It seems to me it would nicely fit as a regular test in the generic loop over distributions https://github.com/scipy/scipy/blob/master/scipy/stats/tests/test_continuous_basic.py#L133

Reply:
Because Ralf told me to [here](https://github.com/scipy/scipy/issues/5982#issuecomment-770967516).
Thanks for taking a look. I'm going to refractor a little bit to simplify. I'll ping you when that's done?

Reply:
Ok, would not be my preference to put it here, but hey --- you guys are doing it, and I'm not, so fine by me :-). 

Sure thing, feel free to ping me.

Reply:
My suggestion was based on the type of test/check this is, parsing docstrings. That seemed to be a bit unusual/fragile/expensive for the test suite.

The `inspect.signature` way would fit more naturally in the test suite itself I'd think.

Reply:
Yes, and before gh-13312, `numpydoc.docscrape` was not even available in the main test suite. So with this method, it makes sense for it to be here.